File: codes/java/leetcodes/src/main/java/com/hit/basinfo/sort_algorithm/BubbleSort.java
package com.hit.basinfo.sort_algorithm;

/**
 * author:Charies Gavin
 * date:2019/1/7,14:20
 * https:github.com/guobinhit
 * description: Bubble Sort
 */
public class BubbleSort {
    /**
     * Bubble Sort
     *
     * @param nums pending array
     */
    public void bubbleSort(int[] nums) {
        // check parameters
        if (nums != null && nums.length > 1) {
            /**
             * Starting from nums.length - 1, the number of comparisons decreases successively,
             * The first time is nums. length - 1 times, the second time is nums. length - 2 times ...
             */
            for (int i = nums.length - 1; i > 0; i--) {
                for (int j = 0; j < i; j++) {
                    /**
                     * If the former element is larger than the latter,
                     * the order of the two elements is interactive
                     */
                    if (nums[j] > nums[j + 1]) {
                        int temp = nums[j];
                        nums[j] = nums[j + 1];
                        nums[j + 1] = temp;
                    }
                }
            }
        }
    }
}

File: codes/python/leetcodes/src/main/python/com/hit/basinfo/sort_algorithm/BubbleSort.py
"""
 author: Charies Gavin
 date: 11/11/19,10:40 AM
 https:github.com/guobinhit
 description: Bubble Sort
"""


def bubble_sort(nums):
    if len(nums) < 2:
        return

    for i in range(len(nums)):
        for j in range(0, len(nums) - i - 1):
            if nums[j] > nums[j + 1]:
                temp = nums[j]
                nums[j] = nums[j + 1]
                nums[j + 1] = temp
